examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,85 @@
+"""
+Regions extraction using Canonical ICA maps
+===========================================
+
+This example shows how to extract each connected ICA map (mostly a 4D Nifti
+image/object) into a seperate brain activation regions and extracts timeseries
+signals from each seperated region. Both can be done at the same time
+using :class:`nilearn.regions.region_extractor`.
+
+We used 20 resting state functional datasets and
+:class:`nilearn.decomposition.canica` for 4D components.
+
+This example in particular can be used to study functional connectomes using
+correlation network matrices between regions.
+
+Please see the related documentation of :class:`nilearn.regions.region_extractor`
+for more details.
+"""
+import numpy as np
+from nilearn import datasets
+print(" -- Fetching ADHD resting state functional datasets -- ")
+adhd_dataset = datasets.fetch_adhd(n_subjects=20)
+func_filenames = adhd_dataset.func
+confounds = adhd_dataset.confounds
+
+from nilearn.decomposition.canica import CanICA
+print(" -- Canonical ICA decomposition of functional datasets -- ")
+# Initialize canica parameters
+n_components = 10
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache", memory_level=2,
+                threshold=3., random_state=0)
+
+canica.fit(func_filenames)
+components_img = canica.masker_.inverse_transform(canica.components_)
+
+# Segementation step: Region extraction from ICA maps
+# Signals step: Average timeseries signal extraction
+# Both are done by calling fit_transform()
+from nilearn.regions import region_extractor
+print(" -- Extracting regions from ICA maps and timeseries signals -- ")
+extractor = region_extractor.RegionExtractor(
+    components_img, standardize=True, threshold=0.5, min_size=400)
+extractor.fit_transform(func_filenames, confounds=confounds)
+
+regions_extracted = extractor.regions_
+n_regions = regions_extracted.shape[3]
+print(" ====== Regions extracted ====== ")
+print(" -- Number of regions extracted from %d ICA components are %d -- " % (
+    n_components, n_regions))
+
+# Timeseries signals extracted from all subjects
+subjects_timeseries = extractor.signals_
```

Comment:
Considering the memory required by time series, it is not a good idea to keep them in memory.